mdtraj/core/trajectory.py

```diff
@@ -2052,9 +2049,7 @@ def image_molecules(self, inplace=False, anchor_molecules=None, other_molecules=
         if inplace:
             result = self
         else:
-            result = Trajectory(xyz=self.xyz, topology=self.topology, time=self.time,
-                unitcell_lengths=self.unitcell_lengths, unitcell_angles=self.unitcell_angles)
-
+             result = self[:]
```

Comment:
As elsewhere: You replace the result definition in a couple of spots with what is effectively an identical declaration (but not explicitly specifying unitcell conservation - perhaps would be good to add a comment as to what this achieves?

Reply:
As above: the original definition results in attributes of the `result` object being views, not independent copies, of the original numpy arrays, so future alterations in the original trajectory object (e.g. changes in the coordinates) appear in the supposedly independent copy too. The revised `result = self[:]` syntax results in a proper deep copy and so fully independent objects.